products/rust.md

```diff
@@ -0,0 +1,294 @@
+---
+title: Rust
+category: lang
+tags: mozilla-research
+iconSlug: rust
+permalink: /rust
+alternate_urls:
+-   /rustlang
+-   /rust-lang
+releasePolicyLink: https://rust-lang.github.io/rustup/concepts/channels.html
+eolColumn: Supported
+versionCommand: rustc --version
+releaseDateColumn: true
+
+auto:
+-   git: https://github.com/rust-lang/rust.git
+    regex: ^<major>[1-9]\d*)\.(?<minor>\d+)\.?(?<patch>\d+)?$
+
+# eol(x) = A new stable release is produced every six weeks, like clock-work. Only the latest stable release receives patches. Each new stable release spends the preceding six week period as the Beta release. There's also a new 'nightly' build every night, but I don't propose you list all of those.
```

Comment:
In my eyes this rules says `eol(x) = releaseDate(x+1)` but in the list of releases you list all as `eol: false`.

Reply:
> Only the latest stable release receives patches.

Yeah this is the support under question, right? I've never thought of previous releases being "end of life" but now this is making me wonder...

Reply:
I don't know what the **source** is for the claim that patches can only be applied to the latest stable release.

However, here's an example of a patch that was released [earlier this year](https://blog.rust-lang.org/2023/08/03/Rust-1.71.1.html) and according to the [advisory](https://blog.rust-lang.org/2023/08/03/cve-2023-38497.html) published with it, the only path to mitigation was to update to the new patch release on the latest stable version, i.e. the patch wasn't applied to previous releases.

Reply:
...which does seem to suggest that all releases older than the latest are EOL...?

Reply:
> In my eyes this rules says `eol(x) = releaseDate(x+1)` but in the list of releases you list all as `eol: false`.

Yes, I did leave it as an example from other languages. : do you want me to remove it ?

Reply:
> In my eyes this rules says `eol(x) = releaseDate(x+1)` but in the list of releases you list all as `eol: false`.

as of https://stackoverflow.com/questions/67062157/when-do-rust-editions-such-as-2015-reach-end-of-life

> "An edition of Rust is timeless, the implementation will hit EOL. This from Programming Rust Second Edition,"

Reply:
right, IMO the key takeaway for Rust is that you can ***safely*** incorporate all new releases without worrying about a new feature causing issues. breaking changes only happen between ***editions*** -- and there is no EOL for editions, so there are no EOL dates for Rust.

with the minor caveat that if you're using an old ***release*** you might be missing a patch that was published for a newer release.